### PR TITLE
Refactors InspectorProxy to compare pathname portion of url passed in…

### PR DESCRIPTION
**Pull request by 116-7**
… request to local pathname comparator variables to fix issue with other rightward elements of url such as query or fragment entering the comparison and causing 404 errors for key debugging routes.

## Summary:
A change in Chromium appended the query "?for_tabs" to the /json/list request made by Chrome DevTools to find remote debugger targets.

The current comparison in InspectorProxy.js compares the entire node IncomingMessage url field to the local pathname constants. The issue arises as url can also contain the query and fragment portions so the original comparison of "/json/list" === "/json/list" which resolved as true would become "/json/list?for_tabs" === "/json/list" and evaluate to false ultimately resulting in a 404 for the request.

In summary, all these changes/issues caused remote debugging of Hermes code in React Native apps to become unavailable, greatly impacting developer experience.

## Changelog:

[GENERAL] [FIXED] JS Debugging: Fix inspector-proxy to allow for DevTools requests with query strings

### CODE REVIEW

**Review by huntie — APPROVED** [review]

File: packages/dev-middleware/src/inspector-proxy/InspectorProxy.js [hunk 1]
```diff
@@ -94,12 +94,13 @@ export default class InspectorProxy implements InspectorProxyQueries {
     response: ServerResponse,
     next: (?Error) => mixed,
   ) {
+    const path = url.parse(request.url).pathname;
```
Comment (huntie) [inline packages/dev-middleware/src/inspector-proxy/InspectorProxy.js:97]: From CI, I think there's a formatting fix to apply in this file. Please run `yarn format` from the repo root.

nit: While we're here, we can modernise this call to use the WHATWG URL API.

```suggestion
    const path = new URL(request.url).pathname;
```

Comment (robhogan) [reply packages/dev-middleware/src/inspector-proxy/InspectorProxy.js:97]: Let's use `pathname` as well for clarity.

**Review by robhogan — APPROVED** [review]
Thanks for this @116-7 - this looks eligible for picking back to the next RN71+72 release to me.

File: packages/dev-middleware/src/inspector-proxy/InspectorProxy.js (hunk 1 above)
Comment (116-7) [reply packages/dev-middleware/src/inspector-proxy/InspectorProxy.js]: I have made these changes in the latest commit - please see comment above usage of the WHATWG constructor and confirm that the change is appropriate.

Comment (robhogan) [reply packages/dev-middleware/src/inspector-proxy/InspectorProxy.js]: I might change this back before merging tbh - your comment raises good points (`url` is in many ways a more ergonomic API than `URL`), but you can leave this with us.
